contracts/HealthNodePatient.sol

```diff
@@ -10,19 +10,22 @@ contract PatientNode is AccessControl {
     bytes32 public constant DOCTOR_ROLE = keccak256("DOCTOR_ROLE");
```

Comment:
If we use role base access. Will the doctor not have access to all patients record?
I think the implementation should be having a struct Approvals of address(patient) => address(doctor) => bool that keeps the record of the doctors that the user approved to access their medical history.